examples/nlp/language_modeling/megatron_retro_eval_legacy.py

```diff
@@ -0,0 +1,144 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.  All rights reserved.
+#
```

Comment:
Why do we also have the legacy code ?

Reply:
Because we have an already existing NeMo-native RETRO model, which is planned to be deprecated. The new Mcore RETRO model's inference pipeline is drastically different from this NeMo-native model's inference pipeline. Hence we make the existing inferencing pipeline to `*_legacy.py`. We already did the same for the training pipeline.

Reply:
Can you just add a note saying this will soon be deprecated. (Maybe a rough timeline of when you plan to deprecate it. )

Reply:
Sure. I have added relevant deprecated notes on `*_legacy` files in this PR.
Regarding the timeline for official depreciation, it's hard to determine at the moment because many of new model's code is dependent on the ADLR's team for complete quality check.